src/visualizers/widgets/MonacoEditor/MonacoEditorWidget.js

```diff
@@ -0,0 +1,476 @@
+/*globals define, WebGMEGlobal, _*/
+
+define([
+    'js/Utils/ComponentSettings',
+    'text!./MonacoLanguages.json',
+    'text!MonacoThemes/themelist.json',
+    'Chance',
+    'vs/editor/editor.main',
+    'jquery-contextMenu',
+    'css!./styles/MonacoEditorWidget.css',
+], function (
+    ComponentSettings,
+    MonacoLanguages,
+    ThemeList,
+    Chance
+) {
+    'use strict';
+
+    MonacoLanguages = JSON.parse(MonacoLanguages);
+    ThemeList = JSON.parse(ThemeList);
+
+    const WIDGET_CLASS = 'monaco-editor';
+    const DISPLAY_MENU_CLASS = 'display-menu';
+    const DEFAULT_THEMES = ['vs', 'vs-dark', 'hc-black'];
+    const DEFAULT_COLORS = {
+        'vs': '#fffffe',
+        'vs-dark': '#1e1e1e',
+        'hc-black': '#000000'
+    };
+
+    const AVAILABLE_KEYBINDINGS = ['default', 'vim'];
+
+    const TAB_NAV = `<ul class="nav nav-tabs nav-display-menu">
+                        <li class="pull-right nav-item">
+                        </li>
+                        <li data-role="button" class="pull-right nav-item">
+                            <button class="btn btn-link ${DISPLAY_MENU_CLASS}">
+                               <i class="glyphicon glyphicon-cog"></i>
+                               Display Settings
+                            </button>
+                        </li>
+                     </ul>`;
+
+    const chance = new Chance();
+
+    function MonacoEditorWidget(logger, container, config = {}) {
+        this._logger = logger.fork('Widget');
+        const modelURI = config.modelURI ? config.modelURI.split(/\./).shift() : 'inmemory://code';
+        const extension = config.modelURI ? config.modelURI.split(/\./).pop() : '.py';
+        this.modelURI = modelURI + chance.string({length: 5}) + extension;
```

Comment:
I am actually not convinced we need `chance` (pulling in a new dependency just to generate a short string) as this seems like it could be done pretty easily with just `Math.random()`. Am I missing something?